scripts/Inelastic/CrystalField/fitting.py

```diff
@@ -1228,14 +1201,31 @@ def __add__(self, other):
                     params['sp'+str(x)+'.IntensityScaling'] = self.crystalField.IntensityScaling[x]
             if differentIntensities:
                 warnings.warn('Mismatch between IntensityScaling values of CrystalField objects', RuntimeWarning)
+        # Preserve fixes and ties
+        ties = {}
+        fixes = []
+        for prefix, obj in {'ion0.':self.crystalField, 'ion1.':other}.items():
+            tiestr = obj.function.getTies()
+            if tiestr:
+                for tiepair in [tie.split('=') for tie in tiestr.split(',')]:
+                    ties[prefix + tiepair[0]] = tiepair[1]
+            for par_id in [id for id in range(obj.function.nParams()) if obj.function.isFixed(id)]:
+                parName = obj.function.getParamName(par_id)
+                if obj.background is not None:
+                    parName = parName.split('.')[-1]
+                if parName not in self.crystalField.field_parameter_names:
+                    continue
+                fixes.append(prefix + parName)
```

Comment:
Could this be moved into a method that returns the ties and fixes? 
It would help with readability, and prevent this method from growing as it's now ~150 LOC

Reply:
I agree. This method is far too long ;-)